### ISSUE TITLE
Crashing with segfault after opening feed articles

### ISSUE THREAD

**Issue by chronoscz** [lwindolf/liferea#828]
Liferea 1.12.7 running on Xubuntu 20.04 64-bit is crashing with error after clicking on an article:
```
Segmentation fault (core dumped)
```

Sample feed source http://feeds.feedburner.com/delphicz is crashing with many items like http://feedproxy.google.com/~r/Delphicz/~3/4BA0NE7RWYg/post.aspx "Delphi trik 4: Disable Windows Ghosting".

During execution it also prints "sys:1: Warning: g_string_insert_len: assertion 'len == 0 || val != NULL' failed" but that is done earlier during execution before error itself so that is probably not related.

Running with gdb it will produce backtrace:
```
(gdb) bt
#0  0x000055555558e8a4 in update_state_get_lastmodified ()
#1  0x0000555555588455 in  ()
#2  0x00007ffff4955062 in  () at /usr/lib/x86_64-linux-gnu/libsoup-2.4.so.1
#3  0x00007ffff4955a78 in  () at /usr/lib/x86_64-linux-gnu/libsoup-2.4.so.1
#4  0x00007ffff4955b1a in  () at /usr/lib/x86_64-linux-gnu/libsoup-2.4.so.1
#5  0x00007ffff3c39e8e in g_main_context_dispatch () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#6  0x00007ffff3c3a240 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#7  0x00007ffff3c3a2e3 in g_main_context_iteration () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#8  0x00007ffff3e8dfd5 in g_application_run () at /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
#9  0x000055555557658f in main ()
```

Execution of "liferea --debug-all" shows:
```
DB     : loading item 24927
DB     : loading item 24927
DB     : loading item 24927
DB     : loading itemset for node "vncwrvt"
DB     : loading of itemset finished
PERF   : = htmlview_render_item took 0,001s
HTML   : writing 3248 bytes to HTML view
GUI    : = itemlist selection took 0,024s
PERF   : = itemlist_selection_changed took 0,024s
UPDATE : processing request (https://delphi.cz/syndication.axd?post=a24cbe69-9a9a-4f63-8fb4-f2a7dda6abdb)
NET    : downloading https://delphi.cz/syndication.axd?post=a24cbe69-9a9a-4f63-8fb4-f2a7dda6abdb
> GET /syndication.axd?post=a24cbe69-9a9a-4f63-8fb4-f2a7dda6abdb HTTP/1.1
> Soup-Debug-Timestamp: 1588752761
> Soup-Debug: SoupSession 1 (0x55842ceaa100), SoupMessage 45 (0x55842d3c13b0), SoupSocket 33 (0x55842cef74e0)
> Host: delphi.cz
> Accept: application/atom+xml,application/xml;q=0.9,text/xml;q=0.8,*/*;q=0.7
> Accept-Encoding: gzip, deflate
> User-Agent: Liferea/1.12.7 (Linux; https://lzone.de/liferea/) AppleWebKit (KHTML, like Gecko)
> Connection: Keep-Alive
  
< HTTP/1.1 304 Not Modified
< Soup-Debug-Timestamp: 1588752761
< Soup-Debug: SoupMessage 45 (0x55842d3c13b0)
< Cache-Control: public
< Content-Type: application/rss+xml
< Last-Modified: Mon, 01 Jan 0001 00:00:00 GMT
< ETag: "0"
< Server: Microsoft-IIS/8.5
< Content-Disposition: inline; filename=rss.xml
< X-Powered-By: ASP.NET
< X-Powered-By: ARR/3.0
< Date: Wed, 06 May 2020 08:12:40 GMT
  
NET    : download status code: 304
NET    : source after download: >>>https://delphi.cz/syndication.axd?post=a24cbe69-9a9a-4f63-8fb4-f2a7dda6abdb<<<
NET    : 0 bytes downloaded
Segmentation fault (SIGSEGV)
```

I also verified that same issue is on clean Xubuntu 20.04 64-bit installed in VirtualBox.

**Comment by Leiaz** [comment]
This seems related to the changes in commit a51bc58, which was intended to properly handle 304 responses. Reverting it avoids the crash. The crash happens with some item's comments threads, it looks like it happens when there are no comments and a 304.

**Comment by lwindolf** [comment]
Reproduced.

**Comment by lwindolf** [comment]
First fix for git master prepared. A backport for 1.12 will follow.

**Comment by lwindolf** [comment]
Fix in master is finally done. To be released with 1.13.3.

 As there were no other reports and the change is rather extensive I'd like to refrain from backporting.
